File: src/presenters/dive_step.rs
pub mod dive_step {
    use crate::models::dive_step::dive_step::DiveStep;
    use crate::presenters::presenter;

    pub fn enter_dive_step() -> DiveStep {
        let depth = presenter::read_numeric_i32("Enter depth (m):", 1, 100);
        let time = presenter::read_numeric_i32("Enter time (min):", 1, 60);

        return DiveStep { depth, time };
    }
}

File: BubblesDivePlanner/Models/DiveStep.cs
namespace BubblesDivePlanner.Models
{
    public class DiveStep : IDiveStep
    {
        public DiveStep(byte depth, byte time)
        {
            Depth = AssignDepth(depth);
            Time = AssignTime(time);
        }

        public byte Depth { get; }
        public byte Time { get; }

        private static byte AssignDepth(byte depth)
        {
            return depth switch
            {
                > 100 => 100,
                _ => depth
            };
        }

        private static byte AssignTime(byte time)
        {
            return time switch
            {
                > 60 => 60,
                < 1 => 1,
                _ => time
            };
        }
    }
}